crates/cxx-qt-lib/src/types/qset/generate.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+# SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+# SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+#
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+set -e
+
+SCRIPT=$(realpath "$0")
+SCRIPTPATH=$(dirname "$SCRIPT")
+
+function generate_bridge_primitive() {
+    tee "$SCRIPTPATH/qset_$1.rs" <<EOF
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#[cxx::bridge]
+pub mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qset.h");
+        type QSet_$1 = crate::QSet<$1>;
+    }
+
+    #[namespace = "rust::cxxqtlib1::qset_$1"]
+    unsafe extern "C++" {
```

Comment:
At the moment this is C-style FFI again.
We can simplify this with CXX.
Just use:
``` rust
unsafe extern "C++" {
    #[rust_name="cxx_clear"]
    fn clear(self: &mut QSet_$1);
}
```

Which will generate the clear method right onto QSet.

Example patch:
[0001-Move-clear-to-using-CXX-style-ffi.patch.txt](https://github.com/KDAB/cxx-qt/files/10076572/0001-Move-clear-to-using-CXX-style-ffi.patch.txt)

We might also be able to use template methods for the remaining functions, like qset_clone...

Reply:
Done